app/src/main/java/io/github/sumukhshiv/mdbsocials/FirebaseUtils.java

```diff
@@ -0,0 +1,80 @@
+package io.github.sumukhshiv.mdbsocials;
+
+import android.app.Activity;
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.widget.Toast;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+
+/**
+ * Created by sumukhshivakumar on 3/2/17.
+ */
+
+public class FirebaseUtils {
+
+    public static void login(final String email, String password, final ProgressDialog progressDialog, Activity activity) {
+        FirebaseAuth firebaseAuth = FirebaseAuth.getInstance();
+        final Context context = activity.getApplicationContext();
+        firebaseAuth.signInWithEmailAndPassword(email, password)
+                .addOnCompleteListener(activity, new OnCompleteListener<AuthResult>() {
+                    @Override
+                    public void onComplete(@NonNull Task<AuthResult> task) {
+                        progressDialog.dismiss();
+                        if (task.isSuccessful()) {
+                            Toast.makeText(context, "Login Successful!", Toast.LENGTH_SHORT).show();
+                            Intent loginToFeedIntent = new Intent(context, UserArea.class);
+                            loginToFeedIntent.putExtra("email", email);
+                            loginToFeedIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                            context.startActivity(loginToFeedIntent);
+                        } else if (!task.isSuccessful()) {
+                            Toast.makeText(context, "Email/Password combination is not correct. Please Try Again.", Toast.LENGTH_LONG).show();
+                        }
+
+                    }
+                });
+
+    }
+
+    public static void signUp(final String email, final String password, final ProgressDialog progressDialog, Activity activity) {
+        final FirebaseAuth mAuth = FirebaseAuth.getInstance();
+        final Context context = activity.getApplicationContext();
+        mAuth.createUserWithEmailAndPassword(email, password)
+                .addOnCompleteListener(activity, new OnCompleteListener<AuthResult>() {
+                    @Override
+                    public void onComplete(@NonNull Task<AuthResult> task) {
+                        if (!task.isSuccessful()) {
+
+                            //Various checks to see whether user input is valid
+                            if (!email.contains("@")) {
+                                Toast.makeText(context, "Please enter a valid Email Address", Toast.LENGTH_SHORT).show();
+                            }
+                            else if (mAuth.signInWithEmailAndPassword(email, password) != null) {
+                                Toast.makeText(context, "An account with this email already exists.",
```

Comment:
In this case, the task is not successful but it does not return anything. How would I go about finding why the task failed. I had it this way, so that the user at least has some insight as to why he/she was not able to register.